core/src/main/java/com/cloud/storage/JavaStorageLayer.java

```diff
@@ -178,18 +183,25 @@ public long getSize(String path) {
     }
 
     @Override
-    public File createUniqDir() {
+    public File createUniqDir() throws IOException {
         String dirName = System.getProperty("java.io.tmpdir");
         if (dirName != null) {
             File dir = new File(dirName);
             if (dir.exists()) {
+                if (isWorldReadable(dir)) {
+                    if (dir.getAbsolutePath().equals("/tmp")) {
```

Comment:
What do you think of extracting `"/tmp"` into a constant in this class?

Reply:
/tmp is the default tmp dir for linux. if we directly tell the user "temp dir is is World Readable", we may mislead user to change the permission of "/tmp", which is not desirable. So we just tell user "the temp dir is /tmp" and hope they change the default temp dir.

Reply:
@lujiefsi I think @GabrielBrascher means that we should add a 
```
    private static final String STDTMP = "/tmp";
```
and than 
```suggestion
                    if (STDTMP.equals(dir.getAbsolutePath()) {
```

Reply:
Sorry for not being clear about the idea, @lujiefsi.
Good that @DaanHoogland explained it better than me :slightly_smiling_face:

Reply:
sorry for my mis-undstand